src/components/PostProcessedApiSpec/PostProcessedApiSpec.tsx

```diff
@@ -26,10 +27,48 @@ const PostProcessedApiSpec: React.FC = () => {
       ) : error ? (
         <S.ErrorLabel>{error.message}</S.ErrorLabel>
       ) : (
-        data && <SwaggerUI spec={YAML.parse(data)} plugins={[TableOfContentsPlugin, DynamicServersPlugin]} />
+        data && <SwaggerUI spec={parseSpec(YAML.parse(data))} plugins={[TableOfContentsPlugin, DynamicServersPlugin]} />
       )}
     </S.PostProcessedApiSpecContainer>
   );
 };
 
+const parseSpec = (spec: any) => {
```

Comment:
This looks good to me as I'm thinking it doesn't have to be recursive since you have a fixed amount of levels.

I would maybe recommend moving the `x-kusk` string to a constant.